### PR TITLE
Case 22045: Added support for OpenEXR files (mostly for skyboxes)

### PR DESCRIPTION
**Pull request by Zvork**
https://highfidelity.fogbugz.com/f/cases/22045/Adding-support-for-openEXR-images-on-pc-and-baking-of-hdr-skyboxes-into-hdr-KTX-format

This PR adds support for the [OpenEXR high dynamic range image file format](http://www.openexr.com/). This is of particular interest for sky maps and ambient maps.

See the updated test in this [hifi_test PR #388](https://github.com/highfidelity/hifi_tests/pull/388).

This PR also adds a new image::Image class to encapsulate image data without QImage to handle a common container for both LDR and HDR images unconfusingly.

### CODE REVIEW

File: cmake/ports/hifi-deps/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: hifi-deps
-Version: 0
+Version: 0.1
 Description: Collected dependencies for High Fidelity applications
-Build-Depends: bullet3, draco, etc2comp, glm, nvtt, openssl (windows), tbb (!android&!osx), zlib
+Build-Depends: bullet3, openexr (!android), draco, etc2comp, glm, nvtt, openssl (windows), tbb (!android&!osx), zlib
```
Comment (jherico) [inline cmake/ports/hifi-deps/CONTROL]: the depends here are intended to be sorted.  openexr should go between nvtt and openssl

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18396/HighFidelity-Beta-PR15264-a93a081.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18396/HighFidelity-Beta-Interface-PR15264-a93a081.exe

File: cmake/ports/hifi-deps/CONTROL (hunk 1 above)
Comment (Zvork) [reply cmake/ports/hifi-deps/CONTROL]: What is the sorting criteria, by the way?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18413/HighFidelity-Beta-PR15264-7791401.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18413/HighFidelity-Quest-Beta-PR15264-7791401.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18414/HighFidelity-Beta-PR15264-2a0033a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18414/HighFidelity-Quest-Beta-PR15264-2a0033a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18414/HighFidelity-Beta-PR15264-2a0033a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18414/HighFidelity-Beta-Interface-PR15264-2a0033a.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18415/HighFidelity-Beta-PR15264-89fbe13.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18415/HighFidelity-Quest-Beta-PR15264-89fbe13.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18415/HighFidelity-Beta-PR15264-89fbe13.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18415/HighFidelity-Beta-Interface-PR15264-89fbe13.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18416/HighFidelity-Beta-PR15264-809b169.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18416/HighFidelity-Quest-Beta-PR15264-809b169.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18416/HighFidelity-Beta-PR15264-809b169.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18416/HighFidelity-Beta-Interface-PR15264-809b169.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18419/HighFidelity-Beta-PR15264-610d834.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18419/HighFidelity-Quest-Beta-PR15264-610d834.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18422/HighFidelity-Beta-PR15264-eb0e1d5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18422/HighFidelity-Quest-Beta-PR15264-eb0e1d5.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18419/HighFidelity-Beta-PR15264-610d834.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18419/HighFidelity-Beta-Interface-PR15264-610d834.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18422/HighFidelity-Beta-PR15264-eb0e1d5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18422/HighFidelity-Beta-Interface-PR15264-eb0e1d5.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18440/HighFidelity-Beta-PR15264-c480f36.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18440/HighFidelity-Quest-Beta-PR15264-c480f36.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18440/HighFidelity-Beta-PR15264-c480f36.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18440/HighFidelity-Beta-Interface-PR15264-c480f36.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18440/HighFidelity-Beta-PR15264-c480f36.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18440/HighFidelity-Beta-Interface-PR15264-c480f36.exe

**Comment by Zvork** [comment]
By the way, OpenEXR support is disabled on Android for this PR. Is this ok @samcake ?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18476/HighFidelity-Beta-PR15264-7455ddb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18476/HighFidelity-Quest-Beta-PR15264-7455ddb.apk)

**Comment by Zvork** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18478/HighFidelity-Beta-PR15264-7455ddb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18478/HighFidelity-Quest-Beta-PR15264-7455ddb.apk)

**Comment by Zvork** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18514/HighFidelity-Beta-PR15264-7455ddb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18514/HighFidelity-Quest-Beta-PR15264-7455ddb.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18515/HighFidelity-Beta-PR15264-4e39a06.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18515/HighFidelity-Quest-Beta-PR15264-4e39a06.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18516/HighFidelity-Beta-PR15264-e2bc80b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18516/HighFidelity-Quest-Beta-PR15264-e2bc80b.apk)

**Comment by Zvork** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18545/HighFidelity-Beta-PR15264-e2bc80b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18545/HighFidelity-Quest-Beta-PR15264-e2bc80b.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18546/HighFidelity-Beta-PR15264-a79e7a2.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18546/HighFidelity-Quest-Beta-PR15264-a79e7a2.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18546/HighFidelity-Beta-PR15264-a79e7a2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18546/HighFidelity-Beta-Interface-PR15264-a79e7a2.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18547/HighFidelity-Beta-PR15264-6b876e6.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18547/HighFidelity-Quest-Beta-PR15264-6b876e6.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18547/HighFidelity-Beta-PR15264-6b876e6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18547/HighFidelity-Beta-Interface-PR15264-6b876e6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18547/HighFidelity-Beta-PR15264-6b876e6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18547/HighFidelity-Beta-Interface-PR15264-6b876e6.exe

File: cmake/ports/hifi-deps/CONTROL (hunk 1 above)
Comment (jherico) [reply cmake/ports/hifi-deps/CONTROL]: alphabetical (US english)

**Pull request merged**